### ISSUE TITLE
[Bug]: Storefront is loading perpetually !! 

### ISSUE THREAD

**Issue by ArtOfThings** [metaplex-foundation/metaplex#2043]
### Which package is this bug report for?

storefront

### Issue description

Hello guys,

I have deployed the Metaplex on my domain and I'm having an issue with the homepage loading perpetually. What have I done wrong ?!

Is it a know issue and under fix or is there a ready fix for it?

### Operating system

Windows 10

### Priority this issue should have

High (immediate attention needed)

### Check the Docs First

- [X] I have checked the docs and it didn't solve my issue

**Comment by mikecann** [comment]
Note another long discussion on this here: https://github.com/metaplex-foundation/metaplex/issues/1339

**Comment by ArtOfThings** [comment]
I guess this is a serious issue and needs some action to solve it since my ticket is the 2nd one and the mentioned one isn't resolved!! Apparently a lot of users are suffering from the latest update which needs a fix.

**Comment by jmar777** [comment]
Same issue here. Looking at the web inspector, it seems to be caused by hanging calls to the `getProgramAccounts` jsonrpc endpoint:

<img width="965" alt="Screen Shot 2022-04-15 at 2 53 04 PM" src="https://user-images.githubusercontent.com/183199/163621634-bb3d9384-9af3-4974-a8db-fa5b7a2954a1.png">

**Comment by ArtOfThings** [comment]
> Same issue here. Looking at the web inspector, it seems to be caused by hanging calls to the `getProgramAccounts` jsonrpc endpoint:
> 
> <img alt="Screen Shot 2022-04-15 at 2 53 04 PM" width="965" src="https://user-images.githubusercontent.com/183199/163621634-bb3d9384-9af3-4974-a8db-fa5b7a2954a1.png">

any idea of a temporary fix ?

**Comment by jmar777** [comment]
> any idea of a temporary fix ?

I tried returning an empty object from that method, but that created errors. Since all I was trying to do for now was minting, I then tried using candy machine directly, but attempting to run that with ts-node resulted in a large number of TypeScript errors.


I have quite a bit of experience with TypeScript and node, but unfortunately I just didn't have time to mess around with this right now. After literally an entire day, all I had been able to do was mint one collection parent and one master edition. I ended up burning those, giving up, and went over to holaplex. It's a shame, because I would have rather minted with metaplex directly, but the project doesn't seem to be in a stable state right now.

**Comment by mikecann** [comment]
@jmar777 if you have coding experience I would suggest just using the [sdk](https://docs.metaplex.com/sdk/js/getting-started). Just smash out a quick script and call it a day:

```ts
it(`creates a mint`, async () => {
  const mint = await actions.mintNFT({
    connection,
    wallet: new NodeWallet(test1Wallet),
    uri: pinkPirateMetadataUri,
    maxSupply: 100,
  });

  console.log(`mintNFTResult`, {
    tx: mint.txId,
    mint: mint.mint.toBase58(),
    metadata: mint.metadata.toBase58(),
    edition: mint.edition.toBase58(),
  });

  // Must wait for confirmation before we can go
  await connection.confirmTransaction(mint.txId);

  // We can now create editions from it :)
  const edition = await actions.mintEditionFromMaster({
    connection,
    wallet: new NodeWallet(test1Wallet),
    masterEditionMint: mint.mint,
    updateAuthority: undefined, // IMMUTABLE
  });

  console.log(`edition`, {
    tx: edition.txId,
    mint: edition.mint.toBase58(),
    metadata: edition.metadata.toBase58(),
    edition: edition.edition.toBase58(),
  });

  // Must wait for confirmation before we can go
  await connection.confirmTransaction(edition.txId);
}, 200000);
```

**Comment by jmar777** [comment]
@mikecann this is great, thank you!

**Comment by BlockchainDataMiner** [comment]
If there is a solution what is it ? Thanks.

**Comment by BlockchainDataMiner** [comment]
After a lot of debugging, I have noticed that the issue seems to come from the getProgramAccounts() function in the web3.ts file, and is probably due to the devnet RPC endpoint that seems to be overloaded sometimes : https://api.devnet.solana.com

**Comment by andrejsim** [comment]
This issue persists, is there a way to have feed back from anyone who has resolved the issue? I also use the metaplex platform on Google Cloud Run, is there a possibility that other user have this issue due to deployment on google over local?

Also the solutions outlined in this thread as well as the thread in #1339 are not outlined in the documentation, who can make these adjustments there?

**Comment by Nedi11** [comment]
In my case, I couldn't initialize the store at all but found this temporary solution: https://github.com/metaplex-foundation/metaplex/issues/1339#issuecomment-1108883634

**Comment by LipLoody** [comment]
in mainet the loading is slow too due to https://github.com/metaplex-foundation/metaplex/issues/2085

**Comment by BlockchainDataMiner** [comment]
> in mainet the loading is slow too due to #2085

Yes, and this large amount of data downloaded is due to a lot of calls to processMetaplexAccounts() function...

**Comment by andrejsim** [comment]
I NOTICED THAT INFINITE LOADING FOR THE STOREFRONT CAN BE ADDRESSED IN DEVELOPER TOOL SETTINGS IN YOUR BROWSER BY 'ENABLE JAVA SOURCE MAP'=FALSE. 
This is not something that most non developer users will know, is there a way to address this with an error message or a chrome setting?

**Comment by VerbotenLad** [comment]
Change the RPC endpoint to the genesys go RPC endpoint.
Upgrade Auction Performance in the localhost:portnumber/admin
Reduce the DEFAULT_TIMEOUT in connection.tsx

Look into all the ways to reduce the loading times/speed up a web app and you'll find answers there.

**Comment by ernitingarg09** [comment]
> I NOTICED THAT INFINITE LOADING FOR THE STOREFRONT CAN BE ADDRESSED IN DEVELOPER TOOL SETTINGS IN YOUR BROWSER BY 'ENABLE JAVA SOURCE MAP'=FALSE. This is not something that most non developer users will know, is there a way to address this with an error message or a chrome setting?

this doesn't help !

**Comment by geokomplo** [comment]
Many people are complaining about this problem but it is still not fixed! Like a joke...

**Comment by na3er-faraji** [comment]
and now it's a deprecated repo, it means this bug is not going to be fixed anymore?
https://github.com/metaplex-foundation/deprecated-storefront
